### PR TITLE
perf: add SIMD acceleration for doubleMean aggregator

### PR DESCRIPTION
**Pull request by jtuglu1**
### Description

We have a few use-cases of `doubleMean` internally. This along with min/max should finalize basic coverage of all common aggregators.

#### Benchmarks
| Null pattern | Vector size | Scalar ns/op | SIMD ns/op | Speedup |
|---|---:|---:|---:|---:|
| none | 128 | 577.647 | 42.667 | 13.54x |
| none | 512 | 2252.809 | 198.616 | 11.34x |
| none | 1024 | 4728.788 | 389.558 | 12.14x |
| none | 4096 | 18346.941 | 1503.129 | 12.21x |
| sparse | 128 | 498.490 | 91.306 | 5.46x |
| sparse | 512 | 1991.397 | 374.181 | 5.32x |
| sparse | 1024 | 4158.765 | 729.459 | 5.70x |
| sparse | 4096 | 18250.915 | 2818.945 | 6.47x |
| alternating | 128 | 257.435 | 91.506 | 2.81x |
| alternating | 512 | 1105.482 | 378.452 | 2.92x |
| alternating | 1024 | 2284.373 | 716.764 | 3.19x |
| alternating | 4096 | 8967.622 | 3033.222 | 2.96x |

#### Release note

Add SIMD acceleration for doubleMean aggregator

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 6 of 6 changed files.

---

This is an automated review by Codex GPT-5.6-Sol

File: processing/src/test/java/org/apache/druid/query/aggregation/simd/SimdDoubleMeanVectorAggregatorTest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  {
+    final Random r = new Random(0xC0FFEEL);
+    final double[] out = new double[size];
+    for (int i = 0; i < size; i++) {
+      out[i] = (r.nextDouble() - 0.5) * 1000.0;
+    }
+    return out;
+  }
+
+  private enum NullPattern
```
Comment (clintropolis) [inline processing/src/test/java/org/apache/druid/query/aggregation/simd/SimdDoubleMeanVectorAggregatorTest.java]: #19660 added `SimdAggregatorTestHelpers` which might be useful here too

File: processing/src/main/java/org/apache/druid/query/aggregation/simd/SimdDoubleMeanVectorAggregator.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import org.apache.druid.segment.vector.VectorValueSelector;
+
+import java.nio.ByteBuffer;
+
+/**
+ * SIMD specialization of {@link DoubleMeanVectorAggregator}'s ungrouped contiguous-range aggregation. The hot loop
+ * reduces input values into one local sum and count, then updates the mean holder once. The grouped scatter-gather
+ * variant is inherited from the parent scalar class.
+ */
+public final class SimdDoubleMeanVectorAggregator extends DoubleMeanVectorAggregator implements NullAwareVectorAggregator
```
Comment (clintropolis) [inline processing/src/main/java/org/apache/druid/query/aggregation/simd/SimdDoubleMeanVectorAggregator.java]: I suppose `NullAwareVectorAggregator` doesn't give you much benefit since `DoubleMeanAggregatorFactory` doesn't extend `NullableNumericAggregatorFactory` I think and be less confusing. I think is just a small change, instead of overriding the aggregate method that takes a boolean array of nulls, that would just be a private method

**Review by clintropolis — COMMENTED** [review]
:metal:

**Review by clintropolis — APPROVED** [review]
